models/DataObject/ClassDefinition/Data/Geopolyline.php

```diff
@@ -0,0 +1,367 @@
+                    $points[] = new DataObject\Data\Geopoint($point['longitude'], $point['latitude']);
+                } else {
+                    throw new \Exception('cannot get values from web service import - invalid data');
+                }
+            }
+
+            return $points;
+        } else {
+            throw new \Exception('cannot get values from web service import - invalid data');
+        }
+    }
+
+    /** True if change is allowed in edit mode.
+     * @param string $object
+     * @param mixed $params
+     *
+     * @return bool
+     */
+    public function isDiffChangeAllowed($object, $params = [])
+    {
+        return true;
+    }
+
+    /** Generates a pretty version preview (similar to getVersionPreview) can be either html or
+     * a image URL. See the ObjectMerger plugin documentation for details
+     *
+     * @param $data
+     * @param null $object
+     * @param mixed $params
+     *
+     * @return array|string
+     */
+    public function getDiffVersionPreview($data, $object = null, $params = [])
+    {
+        if (!empty($data)) {
+            $line = '';
+            $isFirst = true;
+            if (is_array($data)) {
+                foreach ($data as $point) {
+                    if (!$isFirst) {
+                        $line .= ' ';
+                    }
+                    $line .= $point->getLatitude() . ',' . $point->getLongitude();
+                    $isFirst = false;
+                }
+
+                return $line;
+            }
+        }
+
+        return;
+    }
+
+    /** Encode value for packing it into a single column.
+     * @param mixed $value
+     * @param Model\DataObject\AbstractObject $object
+     * @param mixed $params
+     *
+     * @return mixed
+     */
+    public function marshal($value, $object = null, $params = [])
+    {
+        if ($value) {
+            $value = Serialize::unserialize($value);
+            $result = [];
+            if (is_array($value)) {
+                /** @var $point DataObject\Data\Geopoint */
+                foreach ($value as $point) {
+                    $result[] = [
+                        $point->getLatitude(),
+                        $point->getLongitude()
+                    ];
+                }
+            }
+
+            return [
+                'value' => json_encode($result)
+            ];
+        }
+    }
+
+    /** See marshal
+     * @param mixed $value
+     * @param Model\DataObject\AbstractObject $object
+     * @param mixed $params
+     *
+     * @return mixed
+     */
+    public function unmarshal($value, $object = null, $params = [])
+    {
+        if ($value && $value['value']) {
+            $value = json_decode($value['value']);
+            $result = [];
+            if (is_array($value)) {
+                foreach ($value as $point) {
+                    $newPoint = new DataObject\Data\Geopoint($point[1], $point[1]);
```

Comment:
`$point[1], $point[1]` should be `$point[0], $point[1]` and I don't think the below two lines are necessary since lat and long are set in the constructor here anyway.